console/frontend/src/main/frontend/src/angularjs/app/views/webservices/webservices.component.ts

```diff
@@ -0,0 +1,29 @@
+import { appModule } from "../../app.module";
+import { ApiService } from "../../services/api.service";
+import { MiscService } from "../../services/misc.service";
+
+class WebservicesController {
+  rootURL: string = "";
+
+  constructor(
+    private Api: ApiService,
+    private Misc: MiscService
+  ) { };
+
+  $onInit() {
+    this.rootURL = this.Misc.getServerPath();
+
+    this.Api.Get("webservices", (data) => {
+      Object.assign(this, data);
+    });
+  };
+
+  compileURL(apiListener: any) {
+    return this.rootURL + "iaf/api/webservices/openapi.json?uri=" + encodeURI(apiListener.uriPattern);
```

Comment:
I would love to see this as an environment variable

Reply:
Great idea for after we've upgraded everything to angular

Reply:
It's so little effort, tho.

Procrastinating on these kinds of things will only add to the refactoring process.

Reply:
It's only used twice, once in the component logic and once in the template
I do agree that it should be in the environment variables however I dont really see it as part of the upgrade process we're doing right now

Reply:
In my opinion, small things like this are next to no work, but can be overlooked in the future. You make a good point, stating that this is not part of the upgrade.